bust-a-move-framework/src/main/java/bustamove/bubble/Bubble.java

```diff
@@ -110,347 +49,154 @@
      */
     public Bubble() {
     }
-
-    /**
-     * Creates Bubble instance.
-     *
-     * @param xPos      double value for starting x position
-     * @param yPos      double value for starting y position
-     * @param c         ColorChoice enum value for bubble's color
-     * @param forCannon boolean describing if Bubble is for
-     *                  Cannon or not (map otherwise)
-     */
-    public Bubble(final float xPos, final float yPos, final ColorChoice c,
-                  final boolean forCannon) {
-        this.pos = new Point(xPos, yPos);
-        this.xSpeed = 0;
-        this.ySpeed = 0;
-        this.color = c;
-        if (forCannon) {
-            this.state = State.NEW;
-        } else {
-            this.state = State.LANDED;
-        }
-        boundingBox = new Circle((float) (pos.getX() + Bubble.DIAMETER / 2),
-                (float) (pos.getY() + Bubble.DIAMETER / 2),
-                (float) Bubble.DIAMETER / 2);
-
-        switch (color) {
-            case RED:
-                this.drawColor = Bubble.RED_COLOR;
-                break;
-            case BLUE:
-                this.drawColor = Bubble.BLUE_COLOR;
-                break;
-            case YELLOW:
-                this.drawColor = Bubble.YELLOW_COLOR;
-                break;
-            case GREEN:
-                this.drawColor = Bubble.GREEN_COLOR;
-                break;
-            default:
-                break;
-        }
-    }
-
     /**
      * Sets the gamehead, this is a instance of GameData.
      * @param game the gamehead to refer to
      */
     //CHECKSTYLE:OFF: DesignForExtension
-    public void setGameHead(final GameData game) {
-      this.gamehead = game;
-    }
+    public abstract void setGameHead(final GameData game);
     //CHECKSTYLE:ON: DesignForExtension
-
     /**
      * Returns the gamehead.
      * @return  GameData gamehead
      */
-     //CHECKSTYLE:OFF: DesignForExtension
-    public GameData getGameHead() {
-      return this.gamehead;
-    }
+    //CHECKSTYLE:OFF: DesignForExtension

```

Comment:
Can remove these lines since it's abstract now.
(Remove the at line 57(setGameHead) and line 72(draw)